appsmithorg/appsmith
# fix: 10771 show text widget warning on exceeding line breaks limit

## Description

Case: Attempting to render a large json object in a TextWidget. Currently, it makes the app unresponsive due large number of line breaks.
Fix: If line breaks > 5000; then render the text without line breaks and show a warning to the user about the same.

Fixes #10771 

https://github.com/appsmithorg/appsmith/issues/10771

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Add a text widget to your app.
2. Fetch data for the text widget using this api: ```https://restcountries.com/v3.1/all```.
3. Observe that the text is rendered without line breaks and that user is shown an appropriate warning in the property pane. (See image attached)

<img width="1680" alt="Screen Shot 2022-03-11 at 8 07 11 PM" src="https://user-images.githubusercontent.com/5424788/158005791-aa015ebd-8a2d-47e7-acb5-e2c5673865e2.png">
<img width="1680" alt="Screen Shot 2022-03-12 at 11 17 44 AM" src="https://user-images.githubusercontent.com/5424788/158005796-bc9e5ef2-6e2b-4f2c-9dcb-e2d99c2f8e19.png">


## Checklist:











## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>


    // Code coverage diff between base branch:release and head branch: fix-10771_text_widget_warning 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.7 **(-0.01)** | 37.09 **(-0.01)** | 35.28 **(-0.01)** | 55.95 **(-0.02)**
 :red_circle: | app/client/src/workers/helpers.ts | 41.94 **(-58.06)** | 12.5 **(-87.5)** | 50 **(-50)** | 36 **(-64)**
 :red_circle: | app/client/src/workers/validations.ts | 80.39 **(-0.93)** | 71.33 **(-0.36)** | 74.19 **(-2.48)** | 81.35 **(-1.04)**</details>
